src/big/big_int.cr

```diff
@@ -153,6 +299,133 @@ struct BigInt < Int
     self * other.to_big_i
   end
 
+  # a.mul(b) a*b avoiding allocations by mutating a
+  #
+  #    a = 100.to_big_i
+  #    a.mul(10_to_big_i) => # a = 1000
+  def mul(other : BigInt) : self
+    LibGMP.mul(self, self, other)
+    self
+  end
+
+  # a.mul(b) a*b avoiding allocations by mutating a
+  def mul(other : LibGMP::IntPrimitiveSigned) : self
+    LibGMP.mul_si(self, self, other)
+    self
+  end
+
+  #ditto
```

Comment:
Please format the code using `crystal tool format`.

Reply:
Yeah, I noticed. Will do.